components/builder-api-client/src/builder.rs

```diff
@@ -369,22 +378,23 @@ impl BuilderAPIProvider for BuilderAPIClient {
 
         let url = format!("rdeps/{}", ident);
 
-        let mut res = self.0
-                          .get_with_custom_url(&url, |u| {
-                              u.set_query(Some(&format!("target={}", &target.to_string())))
-                          })
-                          .send()
-                          .map_err(Error::HyperError)?;
+        let mut resp = self.0
+                           .get_with_custom_url(&url, |u| {
+                               u.set_query(Some(&format!("target={}", &target.to_string())))
+                           })
+                           .send()?;
+        debug!("Status: {:?}", resp.status());
 
-        if res.status != StatusCode::Ok {
-            return Err(err_from_response(res));
-        }
+        if resp.status() != StatusCode::OK {
+            Err(err_from_response(resp))
+        } else {
+            let mut encoded = String::new();
+            resp.read_to_string(&mut encoded).map_err(Error::IO)?;
+            trace!("Body: {:?}", encoded);
```

Comment:
Good call on moving this to `trace!`.

Reply:
An idea just occurred to me for this stuff. It would also be something that justifies its own PR, but I figured I'd mention it.

One interesting thing we _might_ want to consider here (at least until we can move our logging over to some kind of tagged / structured logging system, that is), is to add a target to the log call, like so:

```rust
trace!(target: "builder_api_traffic::fetch_rdeps::body", "{:?}", encoded);
```
If you don't explicitly set a target, it defaults to the module you're currently in (thus the module-like `::`-delimited path). _This_ is what you're actually operating on when you do something like `RUST_LOG=foo::bar::baz=trace`.

However, nothing stops you from adding your own targets. This basically lets us establish our own custom parallel logging hierarchy. If you want to see all traffic at trace level, use `builder_api_traffic=trace`. If you only want to see the info for this particular API call, narrow it to `builder_api_traffic::fetch_rdeps`. I added `::body` as an example of how to further refine, but that's not needed.

Ideally, we'd have structured logging so that we could just annotate what's happening here (tack on IDs, status codes, header information, responses, etc.). Depending on how things are actually happening under the hood, we might already be able to use [log_mdc](https://github.com/sfackler/rust-log-mdc) for this (which I've been using on some other in-progress work; however using it with futures requires a bit of extra consideration). However, until we get there, we could use this parallel hierarchy technique to squeeze a bit more functionality and usefulness out of the base logging facilities to extract more information out of our code. 

If we were to go this route, I'd recommend putting the logic behind some helper functions, both to ensure we're logging consistently and with the same overall structure, and to ease a future transition to a real structured logging setup.

Reply:
Interesting idea... I didn't realize you could add a target to the trace command... adding that could indeed be very useful... lets start with that and we can discuss the future state of logging separately